lib/utmp.c

```diff
@@ -152,48 +152,55 @@ ATTR_MALLOC(free)
 static /*@null@*/ /*@only@*/struct utmpx *
 get_current_utmp(pid_t main_pid)
 {
-	struct utmpx  *ut;
-	struct utmpx  *ut_by_pid  = NULL;
-	struct utmpx  *ut_by_line = NULL;
+	enum ut_match {
+		UT_NO_MATCH = 0,
```

Comment:
If you really prefer "boolean" then use "false" here and in the initialisation.
Otherwise initialising enum with integer makes code less readable.

`UT_NO_MATCH` is boolena enough by itself.
Checking `if (match)` is too far from initialisation so using explicit boolean value as initial value will not help.

Reply:
No, I prefer boolean-like behavior of integers in this case.  `false` would be inappropriate.